Update the name of Zestimate sensors

dreed47 opened the pull request:
## Breaking Change:
This is a breaking change as it will change entity_id of Zestimate sensors.  If you have automations relying on your Zestimate sensor(s) you may need to revisit them to fix the ID's to the new ones.   Also, you may need to update your Zestimate sensor(s) in your Lovelace UI.  

## Description:
Changed name property to return Zestimate
and the property address.  This will make it easier
to distinguish multiple Zestimate sensor entities
in the UI.

Also removed MIN_TIME_BETWEEN_UPDATES in
favor of SCAN_INTERVAL per suggestion from
amelchio#9580 on Discord

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>
fixes #23757 

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [ ] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

dreed47 committed:
Zestimate: fix for issue #23757
Changed name property to return Zestimate
and the property address.  This will make it easier
distinguish multiple Zestimate sensor entities
in the UI.

Also removed MIN_TIME_BETWEEN_UPDATES in
favor of SCAN_INTERVAL per suggestion from
amelchio#9580 on Discord

Kane610 commented:
Wouldn't this be a breaking change? Entities would get new ids

dreed47 commented:
> Wouldn't this be a breaking change? Entities would get new ids

It didn't break mine when I was testing.  It just renamed them.  What determines what ID's get assigned?

Kane610 commented:
If the entity id change its a breaking change. If entities has unique id it would keep the same if regardless of name

dreed47 commented:
> If the entity id change its a breaking change. If entities has unique id it would keep the same if regardless of name

yes it is a breaking change.  I updated the pull pull request to indicate it.

thx

andrewsayre commented on homeassistant/components/zestimate/sensor.py:
```diff
@@ -38,10 +37,7 @@
-# Return cached results if last scan was less then this time ago.
-MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
-
+SCAN_INTERVAL = timedelta(hours=24)
```
This appears to be unrelated to the referenced issue.  Please don't combine unrelated changes in bug fixes PRs.

dreed47 replied:
I've updated the breaking changes description to include more details.  Let me know if you think more explanation is needed.  

Would you like me to break out the SCAN_INTERVAL change in to another PR?

andrewsayre reviewed (dismissed):
Please also rewrite the breaking changes description from the end-user's point of view -- what does this mean to them and what must they do change.

balloob commented:
It's not a breaking change if entities have a unique ID.

andrewsayre commented:
@balloob they don’t have a unique id, there’s a previous PR to add it.

dreed47 committed:
Zestimate fix for issue #23757

Changed name property to return Zestimate
and the property address.  This will make it easier
distinguish multiple Zestimate sensor entities
in the UI.

dreed47 committed:
Changed name property to return Zestimate
and the property address.  This will make it easier
distinguish multiple Zestimate sensor entities
in the UI.

balloob commented on homeassistant/components/zestimate/sensor.py:
```diff
@@ -70,7 +70,7 @@ def __init__(self, name, params):
     def unique_id(self):
         """Return the ZPID."""
-        return self.params['zpid']
+        return '{} {}'.format(self._name, self.address)
```
We shouldn't rely on name for a unique ID, names can change.

dreed47 replied:
oops, yeah, my bad.  thats what I get for being in a hurry.  Ill fix and re-push

thx

dreed47 committed:
moved code fix to the correct function

dreed47 committed:
removed code change from unique_id function
